File: it/versions/structures/default/dart/bar/v2_0.dart
class Other {
  String name2;

  Other(
    this.name2
  );

  static Other decode(dynamic _data_dyn) {
    if (!(_data_dyn is Map<String, dynamic>)) {
      throw 'expected Map<String, dynamic>, but got: $_data_dyn';
    }

    Map<String, dynamic> _data = _data_dyn;

    var name2_dyn = _data["name2"];
    if (name2_dyn == null) {
      throw "expected value but was null";
    }
    if (!(name2_dyn is String)) {
      throw 'expected String, but was: $name2_dyn';
    }
    final String name2 = name2_dyn;

    return Other(name2);
  }

  Map<String, dynamic> encode() {
    Map<String, dynamic> _data = Map();

    _data["name2"] = this.name2;

    return _data;
  }
}

File: it/versions/structures/default/python/bar/v2_0.py
class Other:
  def __init__(self, _name2):
    self._name2 = _name2

  @property
  def name2(self):
    return self._name2

  @staticmethod
  def decode(data):
    f_name2 = data["name2"]

    if not isinstance(f_name2, unicode):
      raise Exception("not a string")

    return Other(f_name2)

  def encode(self):
    data = dict()

    if self._name2 is None:
      raise Exception("name2: is a required field")

    data["name2"] = self._name2

    return data

  def __repr__(self):
    return "<Other name2:{!r}>".format(self._name2)
